examples/type-conversion/type_conversion.bal

```diff
@@ -1,71 +1,106 @@
 
-    boolean|error b = boolean.convert(s3);
-
-    float|error af = float.convert(a);
-
-    if (intVal1 is int) {
-        io:println("Int value 1 : " + intVal1);
+    // `string` to `int` conversion is unsafe, since the `string` may not be convertible to `int`.
+    int|error res1 = int.convert(s1);
+    if (res1 is int) {
+        io:println("int value: ", res1);
     } else {
-        io:println("Error: " + intVal1.reason());
+        io:println("error: ", res1.detail().message);
     }
 
-    if (intVal2 is int) {
-        io:println("Int value 2 : " + intVal2);
+    res1 = int.convert(s2);
+    if (res1 is int) {
+        io:println("int value: ", res1);
     } else {
-        io:println("Error: " + intVal2.reason());
+        io:println("error: ", res1.detail().message);
     }
 
-    io:println("Int value 3 : " + intVal3);
+    // A `float` to `int` conversion can result in some of the information getting lost.
+    // However, this conversion is always safe since there is a corresponding `int` representation
+    // for all `float` values except for `NaN` or `infinite` float values, in which case the
+    // the conversion attempt will result in a panic.
+    int intVal = int.convert(f);
+    io:println("int value: ", intVal);
 
-    if (b is boolean) {
-        io:println("Boolean value : " + b);
-    } else {
-        io:println("Error: " + b.reason());
-    }
+    // A `string` to `boolean` conversion is always safe. The `string` value `true` (ignoring case)
+    // evaluates to the `boolean` value `true`, while any other `string` is converted to the
+    // `boolean` value `false`.
+    boolean b = boolean.convert(s3);
+    io:println("boolean value: ", b);
 
-    if (af is float) {
-        io:println("Float value : " + af);
+    // A simple basic typed value held in an `any` typed variable can also be converted to its inherent
+    // type using the `.convert()` method. This attempt is unsafe, since the value may not be compatible
+    // with the target type.
+    float|error res2 = float.convert(a);
+    if (res2 is float) {
+        io:println("float value: ", res2);
     } else {
-        io:println("Error: " + af.reason());
+        io:println("error: ", res2.detail().message);
     }
 }
 
 public function main() {
-    Employee emp = {name: "Jack Sparrow", age: 54, empNo: 100};
-    convertType(emp);
-    convertSimpleTypes();
+    // Attempt converting an `anydata` typed record to another `anydata` typed record.
+    Employee emp = { name: "Jack Sparrow", age: 54, empNo: 100 };
+    convertEmployeeToPerson(emp);
+
+    // Attempt converting an `any` constrained map to an `anydata` typed record.
+    // This conversion would be successful since all the expected elements are present
+    // and no non-`anydata` elements are added to the map.
+    map<any> m = { name: "Jack Sparrow", age: 54, empNo: 100 };
+    convertAnyMapToPerson(m);
+
+    // Add a non-`anydata` `typedesc` element to the map and re-attempt conversion.
```

Comment:
shall we rephrase to something like "Add a non-`anydata` element (e.g., `typedesc`) to..."